File: StartHere/src/main/java/com/lambdaschool/starthere/services/AuthorServiceImpl.java
package com.lambdaschool.starthere.services;

import com.lambdaschool.starthere.models.Author;
import com.lambdaschool.starthere.repository.AuthorRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.ArrayList;

@Service(value = "authorService")
public class AuthorServiceImpl implements AuthorService{

    @Autowired
    AuthorRepository authorRepository;

    @Override
    public ArrayList<Author> findAll() {
        ArrayList<Author> outAuthors = new ArrayList<>();
        authorRepository.findAll().iterator().forEachRemaining(outAuthors::add);
        return outAuthors;
    }
}

File: sprint14/src/main/kotlin/com/lambdaschool/starthere/services/AuthorServiceImpl.kt
package com.lambdaschool.starthere.services

import com.lambdaschool.starthere.models.Author
import com.lambdaschool.starthere.repository.AuthorRepository
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service

import java.util.ArrayList
import java.util.function.Consumer

@Service(value = "authorService")
open class AuthorServiceImpl : AuthorService {

    @Autowired
    internal var authorRepository: AuthorRepository? = null

    override fun findAll(): ArrayList<Author> {
        val outAuthors = ArrayList<Author>()
        authorRepository!!.findAll().iterator().forEachRemaining(Consumer<Author> { outAuthors.add(it) })
        return outAuthors
    }
}
